src/mixins/__tests__/panel.test.js

```diff
@@ -112,13 +112,19 @@ describe('createPanelConsumerMixin', () => {
     expect(emitter.emit).toHaveBeenCalledTimes(1);
     expect(emitter.emit).toHaveBeenLastCalledWith(PANEL_CHANGE_EVENT, false);
 
-    // ↓ this should be replaceable with `wrapper.setData()` but it didn't
-    // trigger the watcher in `createPanelConsumerMixin`.
-    // It's probably a bug from vue-test-utils.
-    // https://github.com/vuejs/vue-test-utils/issues/1756
-    // https://github.com/vuejs/vue-test-utils/issues/149
-    wrapper.vm.$set(wrapper.vm, 'state', { attributeName: true });
-    await Vue.nextTick();
+    if (isVue3) {
+      await wrapper.setData({ state: { attributeName: true } });
+    } else {
+      // ↓ this should be replaceable with `wrapper.setData()` but it didn't
+      // trigger the watcher in `createPanelConsumerMixin`.
+      // It's probably a bug from vue-test-utils.
+      // https://github.com/vuejs/vue-test-utils/issues/1756
+      // https://github.com/vuejs/vue-test-utils/issues/149
+      wrapper.vm.$set(wrapper.vm, 'state', {
```

Comment:
does this not work in vue 3? I'd rather have less conditional code in the tests

Reply:
It works with vue 3 + VTU 2 but it doesn't with vue 2 + VTU 1 (due to VTU 1's bug). The panel is watching `state` (https://github.com/algolia/vue-instantsearch/blob/a43c889402532f9bfda816fed1ffc45a9225781d/src/mixins/panel.js#L59:L60) and that wathcer is not triggered in VTU 1. So unfortunately we have the conditional codes in the tests for the panel. We could just go for `$set` for both vue versions if you want.

Reply:
I think $set for both is better, leaving the comment here noting that maybe in the future it can be setData, I don't think triggering change in a different way based on a conditional is so safe in tests

Reply:
oh, I just tried it and in vue 3, we don't have `.$set` :(